src/transformer/events/core/calendar_subscription_updated.php

```diff
@@ -0,0 +1,81 @@
+<?php
+// This file is part of Moodle - http://moodle.org/
+//
+// Moodle is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// Moodle is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with Moodle.  If not, see <http://www.gnu.org/licenses/>.
+
+/**
+ * Transform for calendar subscription updated event.
+ *
+ * @package   logstore_xapi
+ * @copyright Daniel Bell <daniel@yetanalytics.com>
+ *            Milt Reder <milt@yetanalytics.com>
+ * @license   https://www.gnu.org/copyleft/gpl.html GNU GPL v3 or later
+ */
+
+namespace src\transformer\events\core;
+
+use src\transformer\utils as utils;
+use src\transformer\utils\get_activity as activity;
+
+/**
+ * Transformer fn for calendar subscription updated event
+ *
+ * @param array $config The transformer config settings.
+ * @param \stdClass $event The event to be transformed.
+ * @return array
+ */
+
+function calendar_subscription_updated(array $config, \stdClass $event) {
+    $repo = $config['repo'];
+    $user = $repo->read_record_by_id('user', $event->userid);
+    $course = $event->courseid == 0 ? null : $repo->read_record_by_id('course', $event->courseid);
+    $lang = is_null($course) ? $config['source_lang'] : utils\get_course_lang($course);
+    $subscription = $repo->read_record_by_id('event_subscriptions', $event->objectid);
+    $object_id = $config['app_url'].'/calendar/subscription?id='.$subscription->id;
```

Comment:
i thought we were mostly centralizing id logic inside the object creators. Seems like potential for conflict if not.

Reply:
yeah I can refactor that out

Reply:
https://github.com/yetanalytics/moodle-logstore_xapi/pull/67/commits/12c87c067eaaf65f9f31732c9622e96d71383128